src/client.js

```diff
@@ -1,52 +1,58 @@
 /* eslint-disable no-var, key-spacing, object-curly-spacing, prefer-arrow-callback, semi, keyword-spacing */
 
-/**
- * @param {number} c Total number of hydration islands
- */
-function initPreactIslands(c) {
-	var el = document.currentScript.parentNode;
-	if (!document.getElementById('praect-island-style')) {
-		var s = document.createElement('style');
-		s.id = 'preact-island-style';
-		s.textContent = 'preact-island{display:contents}';
-		document.head.appendChild(s);
-	}
-	var o = new MutationObserver(function (m) {
-		for (var i = 0; i < m.length; i++) {
-			var added = m[i].addedNodes;
-			for (var j = 0; j < added.length; j++) {
-				if (added[j].nodeType !== 1) continue;
-				var id = added[j].getAttribute('data-target');
-				var target = document.querySelector('[data-id="' + id + '"]');
-				if (target) {
-					while (target.firstChild !== null) {
-						target.removeChild(target.firstChild);
-					}
-					while (added[j].firstChild !== null) {
-						target.appendChild(added[j].firstChild);
+function initPreactIslandElement() {
+	class PreactIslandElement extends HTMLElement {
+		connectedCallback() {
+			if (!this.isConnected) return;
+
+			let i = this.getAttribute('data-target');
+			if (!i) return;
+
+			var d = this;
+			function f(el) {
```

Comment:
Is there a better way to get HTML comments than this?

Reply:
Ok, this *should* be more performant now: https://github.com/preactjs/preact-render-to-string/pull/260/files#diff-627fe08d8bba6a4fa76e7c5ed36ef6f16d126733e65e19236b391de0d34f1fe0R14

Decided to go with a NodeIterator. TIL

Reply:
If we are worried about perf we can switch to TreeWalker, but NodeIterator has better browser support if I remember right: 
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/12063586/200042992-66c9df63-6433-4c2c-8806-1488d5017e92.png">

Reply:
More benchmarks if we want to switch:
Chome:
```
Name         Operations per second    Average time, ms
walker       1.0 x 10^5               0.01                ==============================>
iterator     5.0 x 10^4               0.02                ===============>
recursion    7.7 x 10^3               0.13                ==>
```

Safari:
```
Name         Operations per second    Average time, ms (main.ts, line 70)
walker       9.1 x 10^4               0.01                ==============================>
iterator     6.3 x 10^4               0.02                =====================>
recursion    3.6 x 10^3               0.28                =>
```

Reply:
Here's the source for the benchmark:
```ts
import Benchmark from "micro-benchmark";

function nest(_parent: HTMLElement, content: string, n: number) {
  let _el = document.createElement("div");
  _el.innerText = content;
  _parent.appendChild(_el);

  if (--n) nest(_el, content, n);
  else {
    _el.appendChild(document.createComment("FIND_ME"));
  }
}

nest(document.body, "👻", 2000);

function findCommentRecursive(node: ChildNode, text: string): Node | null {
  if (node.nodeType === Node.COMMENT_NODE && node.nodeValue === text) {
    return node;
  }

  let child = node.firstChild;
  while (child) {
    const found = findCommentRecursive(child, text);
    if (found) return found;
    child = child.nextSibling;
  }

  return null;
}

function findCommentNodeIterator(node: ChildNode, text: string): Node | null {
  const iterator = document.createNodeIterator(node, NodeFilter.SHOW_COMMENT);
  let current = iterator.nextNode();
  while (current) {
    if (current.nodeValue === text) return current;
    current = iterator.nextNode();
  }
  return null;
}

function findCommentTreeWalker(node: ChildNode, text: string): Node | null {
  const walker = document.createTreeWalker(node, NodeFilter.SHOW_COMMENT);
  let current = walker.nextNode();
  while (current) {
    if (current.nodeValue === text) return current;
    current = walker.nextNode();
  }
  return null;
}

let result = Benchmark.suite({
  specs: [
    {
      name: "recursion",
      fn: () => {
        findCommentRecursive(document.body, "FIND_ME");
      },
    },
    {
      name: "iterator",
      fn: () => {
        findCommentNodeIterator(document.body, "FIND_ME");
      },
    },
    {
      name: "walker",
      fn: () => {
        findCommentTreeWalker(document.body, "FIND_ME");
      },
    },
  ],
});

let report = Benchmark.report(result);
console.log(report);
```